lib/gpprmsya.gi

```diff
@@ -272,15 +272,39 @@ function ( G )
     return classes;
 end);
 
+BindGlobal("CheapIsomSymAlt",function(a,b)
+local hom;
+  if IsPermGroup(a) then
+    hom:=SmallerDegreePermutationRepresentation(a);
+    if Image(hom)=b then return hom;
+    elif NrMovedPoints(Range(hom))<NrMovedPoints(a) then
+      return hom*IsomorphismGroups(Image(hom),b);
+    else
+      return IsomorphismGroups(a,b);
+    fi;
+  fi;
+end);
+
 InstallMethod( IsomorphismFpGroup, "alternating group", true,
-    [ IsNaturalAlternatingGroup ], 10, # override `IsSimpleGroup' method
+    [ IsAlternatingGroup ], 
+    40, # override `IsSimple...' and `IsNatural...' method
 function(G)
   if Size(G)=1 then TryNextMethod();fi;
   return IsomorphismFpGroup(G,
-           Concatenation("A_",String(Length(MovedPoints(G))),".") );
+           Concatenation("A_",String(AlternatingDegree(G)),".") );
 end);
 
 InstallOtherMethod( IsomorphismFpGroup, "alternating group,name",
+    true,
+    [ IsAlternatingGroup, IsString ],
+    35, # override `IsSimple...' and `IsNatural...' method
+function(G,str)
+local H;
+  H:=AlternatingGroup(AlternatingDegree(G));
+  return CheapIsomSymAlt(G,H)*IsomorphismFpGroup(H,str);
+end);
+
+InstallOtherMethod( IsomorphismFpGroup, "natural alternating group,name",
     true,
     [ IsNaturalAlternatingGroup, IsString ],
```

Comment:
Why is this method kept, isn't the method above always invoked over it?

Reply:
No, It still ranks higher. I welcome suggestions to make this cleaner.

Reply:
I don't understand why it ranks higher? I am seeing this:
```
gap> RankFilter(IsNaturalAlternatingGroup);
65
gap> RankFilter(IsAlternatingGroup);
37
gap> RankFilter(IsAlternatingGroup) + 40 > RankFilter(IsNaturalAlternatingGroup);
true
```

Reply:
I think it is implied filters (but I did not check that). I checked with `ApplicableMethod` and got (excerpt)

    #I  Method 2 : ``IsomorphismFpGroup: natural alternating group,name'' at /Users/hulpke/gap4/lib/gpprmsya.gi:306, value: 78
    #I   - 1st argument needs
    [ "IsNaturalAlternatingGroup", "HasIsNaturalAlternatingGroup" ]    
    #I  Method 3: ``IsomorphismFpGroup: alternating group,name'' at     /Users/hulpke/gap4/lib/gpprmsya.gi:296, value: 77
    #I   - 1st argument needs [ "IsAlternatingGroup", "HasIsAlternatingGroup" ]
    #I  Method 6: ``IsomorphismFpGroup: natural symmetric group,name'' at /Users/hulpke/gap4/lib/gpprmsya.gi:1787, value: 68
    #I   - 1st argument needs
    [ "IsNaturalSymmetricGroup", "HasIsNaturalSymmetricGroup" ]
    #I  Method 7: ``IsomorphismFpGroup: symmetric group,name'' at /Users/hulpke/gap4/lib/gpprmsya.gi:1777, value: 67
    #I   - 1st argument needs [ "IsSymmetricGroup", "HasIsSymmetricGroup" ]

So the ordering is OK. One could go down in the value (I think up to 9) without getting into problems with other methods.

Reply:
The problem is that the difference is not stable -- it is 1 for you, but it is -12 on my system: the difference is heavily impacted by which packages are loaded.

One way to fix this would be to replace the static rank with a dynamic one, such as `{}->RankFilter(IsNaturalAlternatingGroup)-RankFilter(IsAlternatingGroup)+1`

Reply:
Ok -- I did not realize that packages could influence the library that way.

I've changed the code to just have a single method in either case, this should avoid all ambiguity and will also catch the hypothetical case of a group knowing it is alternating but not yet knowing its naturally alternating.